src/components/Asset/AssetActions/Trade/Swap.tsx

```diff
@@ -56,148 +61,189 @@ export default function Swap({
   const [spotPrice, setSpotPrice] = useState<string>()
   const [totalValue, setTotalValue] = useState<string>()
   const [tokenAmount, setTokenAmount] = useState<string>()
+  const [swapFee, setSwapFee] = useState<string>()
 
   useEffect(() => {
-    if (!asset || !balance || !values?.type) return
+    if (!assetExtended || !balance || !values?.type) return
+    const poolInstance = new Pool(web3)
 
     async function calculateMaximum() {
       const amountDataToken =
         values.type === 'buy'
           ? new Decimal(maxDt)
           : new Decimal(balance.datatoken)
-      const amountOcean =
+
+      const amountBaseToken =
         values.type === 'buy'
           ? new Decimal(balance.baseToken)
-          : new Decimal(maxOcean)
-
-      // const maxBuyOcean = await ocean.pool.getOceanReceived(
-      //   price.address,
-      //   `${amountDataToken.toString()}`
-      // )
-      // const maxBuyDt = await ocean.pool.getDTReceived(
-      //   price.address,
-      //   `${amountOcean.toString()}`
-      // )
-
-      // const maximumDt =
-      //   values.type === 'buy'
-      //     ? amountDataToken.greaterThan(new Decimal(maxBuyDt))
-      //       ? maxBuyDt
-      //       : amountDataToken
-      //     : amountDataToken.greaterThan(new Decimal(balance.datatoken))
-      //     ? balance.datatoken
-      //     : amountDataToken
-
-      // const maximumOcean =
-      //   values.type === 'sell'
-      //     ? amountOcean.greaterThan(new Decimal(maxBuyOcean))
-      //       ? maxBuyOcean
-      //       : amountOcean
-      //     : amountOcean.greaterThan(new Decimal(balance.ocean))
-      //     ? balance.ocean
-      //     : amountOcean
-
-      // setMaximumDt(maximumDt.toString())
-      // setMaximumOcean(maximumOcean.toString())
-
-      // setOceanItem((prevState) => ({
-      //   ...prevState,
-      //   amount: amountOcean.toString(),
-      //   maxAmount: maximumOcean.toString()
-      // }))
-
-      // setDtItem((prevState) => ({
-      //   ...prevState,
-      //   amount: amountDataToken.toString(),
-      //   maxAmount: maximumDt.toString()
-      // }))
+          : new Decimal(maxBaseToken)
+
+      try {
+        const swapFee = await poolInstance.getSwapFee(
```

Comment:
we need to make sure this returns same thing as when getting it from subgraph as right now we have 2 different ways of getting swap fee in this PR, output component and in here

Reply:
right, i'll use the poolInfo data in here too